AuthService/models/user.go

```diff
@@ -0,0 +1,35 @@
+package models
+
+import (
+	"encoding/json"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+
+	"github.com/golang-friends/slack-clone/AuthService/configs"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+// NilUser ...
+var NilUser User
+
+// User ...
+type User struct {
+	ID       primitive.ObjectID `bson:"_id"`
+	Username string             `bson:"username"`
+	Password string             `bson:"password"`
+	Email    string             `bson:"email"`
+	Admin    bool               `bson:"admin"`
+}
```

Comment:
How about we define two User models?

- Create a `User` message in the proto
    ```proto
    message User {
        string id = 1;
        string username = 2;
        // ...
    }
    ```
- Rename this to `UserInMongo`
```suggestion
type UserInMongoDb struct {
	ID       primitive.ObjectID `bson:"_id"`
	Username string             `bson:"username"`
	Password string             `bson:"password"`
	Email    string             `bson:"email"`
	Admin    bool               `bson:"admin"`
}
```
- And the rest of business codes only deal with `User` defined in protobuf
- And in the database layer, we define `User <---> UserInMongo` conversions

The reasons are:
- `User` may be used in other services
- There is no tight coupling between DB and the domain model.
  - User as a domain model and how a user is actually stored in database can differ.

Reply:
If we move this to a dynamic db setup, we'll have to adjust the struct name. But for now it's fine.